lib/Demangling/OldDemangler.cpp

```diff
@@ -2229,9 +2234,18 @@ class OldDemangler {
     return true;
   }
 
-  void addImplFunctionAttribute(NodePointer parent, StringRef attr,
-                         Node::Kind kind = Node::Kind::ImplFunctionAttribute) {
-    parent->addChild(Factory.createNode(kind, attr), Factory);
+  void addImplFunctionAttribute(
+      NodePointer parent, StringRef attr, bool forConvention = false,
+      Node::Kind kind = Node::Kind::ImplFunctionAttribute) {
+    if (!forConvention) {
+      parent->addChild(Factory.createNode(kind, attr), Factory);
+      return;
+    }
+    auto attrNode = Factory.createNode(Node::Kind::ImplFunctionAttribute);
+    attrNode->addChild(
+        Factory.createNode(Node::Kind::ImplFunctionConventionName, attr),
+        Factory);
+    parent->addChild(attrNode, Factory);
```

Comment:
If you're going to do this, seems to me that you ought to add a new `ImplFunctionConvention` node kind instead of reusing `ImplFunctionAttribute` with essentially a totally different structure.  That would simplify some of the gnarly logic in remangling / printing these.

Reply:
Added a new `ImplFunctionConvention` node which carries one or two children depending on the exact convention.